tests/test_api.py

```diff
@@ -208,42 +208,42 @@ def test_do_GET(self):
             with patch.object(MockHa, 'is_standby_cluster', Mock(return_value=True)):
                 MockRestApiServer(RestApiHandler, 'GET /standby_leader')
         MockPatroni.dcs.cluster = None
-        with patch.object(RestApiHandler, 'get_postgresql_status', Mock(return_value={'role': 'master'})):
-            MockRestApiServer(RestApiHandler, 'GET /master')
+        with patch.object(RestApiHandler, 'get_postgresql_status', Mock(return_value={'role': 'primary'})):
+            MockRestApiServer(RestApiHandler, 'GET /primary')
         with patch.object(MockHa, 'restart_scheduled', Mock(return_value=True)):
-            MockRestApiServer(RestApiHandler, 'GET /master')
-        self.assertIsNotNone(MockRestApiServer(RestApiHandler, 'GET /master'))
+            MockRestApiServer(RestApiHandler, 'GET /primary')
+        self.assertIsNotNone(MockRestApiServer(RestApiHandler, 'GET /primary'))
```

Comment:
Perhaps should check the content of this query as it's returning `master`

```
postgres@patroni2:~$ curl -s http://127.0.0.1:8008/primary|jq .role
"master"
```

Reply:
Yes, it is expected that new versions will still use "master" for a while.
Right now we change only tests in order to verify that the actual code correctly handles both cases, when role == 'master' or role == 'primary'.
Now we are planning a release 3.0.
My plan is to completely get rid from master only in 3.1, where in release notes we could write - "you should be running at least 3.0.0" in order to safely upgrade to "3.1.0" (mainly because of https://github.com/zalando/patroni/blob/v2.1.5/patroni/ha.py#L42). Upgrading directly from older version is sort of also possible, but there will be very easy to screw things up :(

Reply:
I understand that it is a big change, bumping the major version notifies the user of backwards incompatible changes. Perhaps, if this change is backwards compatible this could go into 2.2, by allowing the user to get used to using the new syntax/interfaces whilst fully supporting all the old, with the main switch to not having `master` in 3.0? That would be more in-line with semver imo.

Reply:
This is not only about breaking the API, which indeed we don't break it yet.
There are two huge features recently merged: DCS failsafe and Citus support.
IMO, it deserves 3.0 :)

Reply:
ah ha! Fair enough.